app/screens/conversion-flow/use-convert-money-details.ts

```diff
@@ -0,0 +1,95 @@
+import { DisplayCurrency, MoneyAmount, WalletOrDisplayCurrency } from "@app/types/amounts"
+import React from "react"
+import { usePriceConversion } from "../../hooks/use-price-conversion"
+import { WalletCurrency } from "@app/graphql/generated"
+
+type Wallet = {
+  balance: number
+  walletCurrency: WalletCurrency
+  id: string
+}
+
+export const useConvertMoneyDetails = () => {
+  const [wallets, _setWallets] = React.useState<{
+    fromWallet: Wallet
+    toWallet: Wallet
+  }>()
+  const [moneyAmount, setMoneyAmount] = React.useState<
+    MoneyAmount<WalletOrDisplayCurrency>
+  >({ amount: 0, currency: DisplayCurrency })
+
+  const setWallets = (wallets: { fromWallet: Wallet; toWallet: Wallet }) => {
+    if (wallets.fromWallet.balance === 0) {
+      return _setWallets({
+        fromWallet: wallets.toWallet,
+        toWallet: wallets.fromWallet,
+      })
+    }
+
+    _setWallets(wallets)
+  }
+
+  const { convertMoneyAmount } = usePriceConversion()
+
+  if (!wallets || !convertMoneyAmount) {
+    return {
+      moneyAmount,
+      setMoneyAmount,
+      setWallets,
+      displayAmount: undefined,
+      settlementSendAmount: undefined,
+      settlementReceiveAmount: undefined,
+      toggleAmountCurrency: undefined,
+      fromWallet: undefined,
+      toWallet: undefined,
+      canToggleWallet: false,
+      toggleWallet: undefined,
+      isValidAmount: false,
+    } as const
+  }
+
+  const { fromWallet, toWallet } = wallets
+
+  const toggleAmountCurrency = () => {
+    setMoneyAmount(
+      convertMoneyAmount(
+        moneyAmount,
+        moneyAmount.currency === DisplayCurrency
+          ? fromWallet.walletCurrency
+          : DisplayCurrency,
+      ),
+    )
+  }
+
+  const toggleWallet = toWallet.balance
```

Comment:
```suggestion
  const toggleWallet = toWallet.balance > 0
```

maybe that would be more explicit about the intent?